sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -817,6 +892,27 @@ public void setValueDeserializer(String valueDeserializer) {
       return new TypedWithoutMetadata<>(this);
     }
 
+    private Schema fetchAvroSchema(String schemaRegistryUrl, String subject) {
+      SchemaRegistryClient registryClient;
+      if (getSchemaRegistryClientFactoryFn() != null) {
+        registryClient = getSchemaRegistryClientFactoryFn().apply(schemaRegistryUrl);
+      } else {
+        registryClient = new CachedSchemaRegistryClient(schemaRegistryUrl, Integer.MAX_VALUE);
+      }
+
+      SchemaMetadata latestSchemaMetadata;
+      try {
+        latestSchemaMetadata = registryClient.getLatestSchemaMetadata(subject);
+      } catch (IOException | RestClientException e) {
+        throw new IllegalArgumentException(
+            "Unable to get latest schema metadata for subject: " + subject, e);
+      }
+
+      final Schema avroSchema = new Schema.Parser().parse(latestSchemaMetadata.getSchema());
+      checkArgument(avroSchema != null, "Avro schema can't be null");
```

Comment:
I guess it's almost impossible but I'd prefer to leave this check "just in case" since we can't be sure on 100% and I don't see any disadvantages to have it here.